### ISSUE TITLE
Support configurable way to log to stdout and to audit log and profile log

### ISSUE THREAD

**Issue by pavan-kalyan** [StarRocks/starrocks#52976]
## Enhancement
Currently sys_log_to_console is the only config that is used to determine if logs should be logged to stdout. If it's false, logs are written to file. If it's true, it's only written to stdout.

We have the need to support dual logging in a configurable way. 
1) We need it to be logged it to stdout and to audit log as well.
2) We need to be able to configure this so that we can support more than audit log in the future. e.g, profile log.

### Proposed solution
The simplest solution is to add a config variable for each type of log file. (based on the current file logging [section](https://github.com/StarRocks/starrocks/blob/bc35b4f98c2ef74872228ca3ee0145824bfde7f7/fe/fe-core/src/main/java/com/starrocks/common/Log4jConfig.java#L161))
1) audit
2) dump
3) big query
4) profile
let me know if I missed any in the list above.

I think the config would be <audit_log>_to_disk_enabled. This would log to disk if sys_log_to_console is true. if sys_log_to_console is false, then this config won't matter because the existing behavior is such that `sys_log_to_console=false` already logs to file, so it would be backward compatible that way as default value of <audit_log>_to_disk_enabled is false.

The reason a separate config for each type of log file is that there are already multipler config options per log type in [Config.java
](https://github.com/StarRocks/starrocks/blob/bc35b4f98c2ef74872228ca3ee0145824bfde7f7/fe/fe-core/src/main/java/com/starrocks/common/Config.java#L150)

I can work on these changes if the proposed plan is fine. Additionally, we want to backport this to 3.3 if possible.

**Comment by pavan-kalyan** [comment]
Can you explain a bit more about how you would approach this? How would this impact all of the existing config variables and how would deployment for docker and helm work in this case?

**Comment by kevincai** [comment]
As of now, there are 40+ configurable items to control the log4j in FE. It can't be sustainable going that way further.

What I am thinking is, to add a new configure item, `log4j_config_file` which points to a log4j.properties file in local filesystem, if this configuration is provided, FE will try to load the log4j.properties as its canonical log configuration and ignore all other log related configurations. So this `log4j.properties` has the full control of log modules, log level, log rotation, log sink, log format, log pattern, e.t.c.. The user will be responsible for providing a valid log4j configuration.

The log4j configuration can be referred in this link: https://logging.apache.org/log4j/2.x/manual/configuration.html

**Comment by pavan-kalyan** [comment]
Yes, I think that makes sense. It works well for our deployment set up too.
Is there an estimated timeline for these changes?

**Comment by dengliu** [comment]
> What I am thinking is, to add a new configure item, log4j_config_file which points to a log4j.properties file in local filesystem, if this configuration is provided, FE will try to load the log4j.properties as its canonical log configuration and ignore all other log related configurations. 

Hi @kevincai  Any ETA of this feature? Thank you!
